model-optimizer/extensions/front/tf/ObjectDetectionAPI.py

```diff
@@ -1185,14 +1198,162 @@ def generate_sub_graph(self, graph: Graph, match: SubgraphMatch):
+        input_image_hw = node_to_get_shape_value_of_indices(input_shape, [1, 2])  # NHWC layout
+        hwhw = create_op_with_const_inputs(graph, Tile, {1: int64_array([2])}, {'name': 'image_hwhw'}, input_image_hw)
+
+        hwhw_float = Cast(graph,
+                          {'dst_type': data_type_str_to_np(graph.graph['cmd_params'].data_type)}).create_node([hwhw])
+        scaled_anchors = Div(graph, {'name': 'scaled_anchors'}).create_node([yxyx_anchors, hwhw_float])
+
+        flattened_anchors = create_op_with_const_inputs(graph, Reshape, {1: int64_array([1, 1, -1])},
+                                                        {'name': 'flattened_anchors'}, scaled_anchors)
+        cropped_anchors = AttributedClamp(graph, {'min': 0.0, 'max': 1.0, 'name': 'clamped_xyxy',
+                                                  'nchw_layout': True}).create_node([flattened_anchors])
+        mark_as_correct_data_layout(flattened_anchors)
+
+        # create tensor of shape [4] with variance values which then are tiled by the number of boxes which is obtained
+        # from the 'yxyx_anchors' node
+        variances = Const(graph, {'value': np.float32([1.0 / variance_x, 1.0 / variance_y, 1.0 / variance_width,
+                                                       1.0 / variance_height])}).create_node()
+
+        anchors_shape = Shape(graph, {'name': 'anchors_shape'}).create_node([yxyx_anchors])
+        anchors_count = node_to_get_shape_value_of_indices(anchors_shape, [0])
+        tiled_anchors = Tile(graph, {'name': 'tiled_anchors'}).create_node([variances, anchors_count])
+        reshaped_tiled_anchors = create_op_with_const_inputs(graph, Reshape, {1: int64_array([1, 1, -1])},
+                                                             {'name': 'flattened_variances'}, tiled_anchors)
+
+        # now we can merge actual anchors coordinates with a tensor with variances as it is expected by the
+        # DetectionOutput operation
+        duplicate_anchors = Concat(graph, {'axis': 1, 'name': 'anchors_with_variances'}).create_node(
+            [cropped_anchors, reshaped_tiled_anchors])
+
+        do = DetectionOutput(graph,
+                             {'background_label_id': 0,
+                              'clip_after_nms': True,
+                              'clip_before_nms': False,
+                              'code_type': 'caffe.PriorBoxParameter.CENTER_SIZE',
+                              'confidence_threshold': 0.0,
+                              'decrease_label_id': False,
+                              'input_height': 1,
+                              'input_width': 1,
+                              'keep_top_k': max_proposals,
+                              'normalized': True,
+                              'num_classes': 2,
+                              'objectness_score': 0,
+                              'share_location': True,
+                              'top_k': 6000,
```

Comment:
Why is it hardcoded? Can we get this number from fw?

Reply:
This number cannot be deduced from the model. This parameter defines number of internal object that will be handled inside the DO operation. So it does not impact the output shapes of the DO operation. The number 6000 was selected rather arbitrary but verified with the customer dataset that it produces correct results on all images. I can hardly imagine that there could be more than 6000 detections per image, so this is a reasonable upper bound which could be applicable for any model.